### PR TITLE
Changes to CLI

### PR DESCRIPTION
**Pull request by howzitcdf**
Small changes to help the developer out when their app is not starting

1. if the cli detects that a function has been passed instead of a Sanic instance it will error out and suggest that the dev uses `--factory` eg. `sanic fake.factory.run`

2. if the cli detects a path has been passed instead of a Sanic instance it will error out and suggest the that the dev uses `--simple` eg. `sanic ./fake/`

I welcome all changes and teachings! :)

### CODE REVIEW

File: sanic/cli/app.py [hunk 1]
```diff
@@ -76,7 +76,12 @@ def run(self):
             kwargs = self._build_run_kwargs()
             app.run(**kwargs)
         except ValueError:
-            error_logger.exception("Failed to run app")
+            if os.path.isdir(self.args.module):
+                error_logger.exception(
+                    "Please attach --simple to run your app from a directory."
+                )
+            else:
```
Comment (ahopkins) [inline sanic/cli/app.py:83]: Why do this check here and not in the `_get_app` exception?

File: sanic/cli/app.py [hunk 2]
```diff
@@ -125,9 +130,18 @@ def _get_app(self):
                 app_type_name = type(app).__name__
 
                 if not isinstance(app, Sanic):
+                    if hasattr(app, "__call__"):
```
Comment (ahopkins) [inline sanic/cli/app.py:133]:
```python
if callable(app):
```

File: sanic/cli/app.py (hunk 1 above)
Comment (howzitcal) [reply sanic/cli/app.py:83]: didn't know we could, let me do that :)

File: sanic/cli/app.py [hunk 3]
```diff
@@ -125,9 +130,18 @@ def _get_app(self):
...
+                        solution = f"sanic {self.args.module} --factory"
+                        raise ValueError(
+                            "Module is not a Sanic app, it is a"
+                            f"{app_type_name}\n"
+                            "  If your function returns a"
```
Comment (ahopkins) [inline sanic/cli/app.py:138]: Let's not presume that it is a function.

```
If this callable returns a
```

File: sanic/cli/app.py (hunk 1 above)
Comment (ahopkins) [reply sanic/cli/app.py:83]: I would just do this check right next to the `--factory` check.

Comment (howzitcal) [reply sanic/cli/app.py:83]: on it :)

File: sanic/cli/app.py (hunk 2 above)
Comment (howzitcal) [reply sanic/cli/app.py:134]: sorted, I learnt something :) thank you!

File: sanic/cli/app.py (hunk 3 above)
Comment (howzitcal) [reply sanic/cli/app.py:139]: done and dusted :)

File: sanic/cli/app.py (hunk 1 above)
Comment (howzitcal) [reply sanic/cli/app.py]: i changed where it raises slightly, however I think it should be fine, let me know :)

**Review by ahopkins — APPROVED** [review]

**Review by ahopkins — APPROVED** [review]

**Pull request merged**